code/game/machinery/vending.dm

```diff
@@ -3706,3 +3706,62 @@ var/global/num_vending_terminals = 1
 		return pick(grey_slogans)
 	else
 		return pick(product_slogans)
+
+/obj/machinery/vending/lotto
+	name = "\improper Lotto Tickets"
+	desc = "Wall-mounted vending machine which dispenses scratch-off lottery tickets. Winners can be cashed at Cargo."
+	product_slogans = list(
+		"Feeling lucky?",
+		"Money won is twice as sweet as money earned.",
+		"The greatest risk is not taking one."
+	)
+	product_ads = list(
+		"Quit while you’re ahead. All the best gamblers do.",
+		"If there weren’t luck involved, I would win every time.",
+		"Better an ounce of luck than a pound of gold.",
+		"Behind bad luck comes good luck."
+	)
+	vend_reply = "Good luck!"
+	icon_state = "Lotto"
+	icon_vend = "Lotto-vend"
+	products = list(
+		/obj/item/toy/lotto_ticket/gold_rush = 20,
+		/obj/item/toy/lotto_ticket/diamond_hands = 20,
+		/obj/item/toy/lotto_ticket/phazon_fortune = 20
+		)
+	contraband = list(
+		/obj/item/toy/lotto_ticket/supermatter_surprise = 1
+		)
+	prices = list(
+		/obj/item/toy/lotto_ticket/gold_rush = 5,
+		/obj/item/toy/lotto_ticket/diamond_hands = 20,
+		/obj/item/toy/lotto_ticket/phazon_fortune = 50,
+		/obj/item/toy/lotto_ticket/supermatter_surprise = 100
+		)
+
+	pack = /obj/structure/vendomatpack/lotto
+
+/obj/machinery/vending/lotto/attackby(obj/item/I as obj, user as mob)
+	add_fingerprint(user)
+	if(istype(I, /obj/item/toy/lotto_ticket))
+		var/obj/item/toy/lotto_ticket/T = I
+		if(!T.iswinner)
+			playsound(src, "buzz-sigh", 50, 1)
+			for(var/mob/V in hearers(src))
+				V.visible_message("<b>[src]</b>'s monitor flashes, \"This ticket is not a winning ticket.\"")
+			return
+		else
+			if(T.winnings <= 10000)
+				for(var/mob/V in hearers(src))
+					V.visible_message("<b>[src]</b>'s monitor flashes, \"Withdrawing [T.winnings] credits from the Central Command Lottery Fund!\"")
+				dispense_cash(T.winnings, get_turf(src))
+				playsound(src, "polaroid", 50, 1)
+			else
+				var/mob/living/carbon/human/H = user
+				var/account_num = H?.mind?.initial_account?.account_number
+				var/datum/money_account/account = get_money_account_global(account_num)
+				account.money += T.winnings
```

Comment:
This is wrong at most levels of analysis:

So you *think* you don't know if the human exists (how are they `attackby`ing this, then?), whether it has a mind, whether that's got an account associated with it, nor whether said account has an account number, YET you don't even bother checking if the account you get from `get_money_account_global()` exists before trying to add money to it.

1) This will result in a runtime error.
2) You *do* know some kind of mob is attacking the vending machine, but you don't know if they're human: you need to check with `istype()`
3) Take a look at what `get_money_account_global()` does: it searches all accounts and returns one matching the provided account number (implicitly returns null otherwise). You don't need to do that, because you already have the account datum in `H.mind.initial_account`.

However, taking a step back and looking at the bigger picture here, why are you using the mob's initial account in the first place? A monkey may very well be using the ticket, or a hobo without an account. If you really want to, you can look at the equipped ID/PDA and send money to that, but I don't see what's wrong with spawning piles of cash.

Reply:
I was having trouble finding how to get a person's account in general, but yeah looking at equipped ID is probably the best way; I can rework that.

As for "[...]but I don't see what's wrong with spawning piles of cash", this is how I initially had it but there were concerns of people stealing. If someone won 1,000,000 credits, there would be 100 stacks of 10x1000 credit bills which would take a significant amount of time to toss into your PDA. Making it spit out cash for all prize amounts is much easier to implement (and is what it used to do) so I would be happy to revert to that approach if you deem theft not to actually be an issue.

Reply:
You could spawn the special admin credit chit that can hold arbitrary amounts of money and set it to the prize value.

Reply:
Resolved in latest commit